cmd/manager/main.go

```diff
@@ -59,6 +62,17 @@ func main() {
 		log.Fatal(err)
 	}
 
+	// Set proxy API availability
+	k8sInterface, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = proxy.SetProxyAvailability(k8sInterface.Discovery())
```

Comment:
are we relying on the pod getting restarted if discovery is temporarily unavailable?

Reply:
Yes - if the discover api fails marketplace would log a fatal error.